### PR TITLE
DOC: Fix some inconsistencies in the docstring of matrix_rank

### PR DESCRIPTION
**Pull request by asmeurer**
- Rename the argument from M to A, since M already refers to the number of
  rows in the docstring.

- Fix the documentation for the formula for the default value of tol. It only
  uses the maximums of the number of rows and columns, not the max of all
  dimensions. The code here was correct; only the documentation was wrong.

### CODE REVIEW

**Comment by charris** [comment]
The test failures are unrelated. The towncrier failures suggest you aren't keeping your main branch current. The lint failures are due to existing long lines, which can be fixed up after this goes in.

File: numpy/linalg/linalg.py [hunk 1]
```diff
@@ -1814,18 +1814,18 @@ def matrix_rank(M, tol=None, hermitian=False):
 
     Parameters
     ----------
-    M : {(M,), (..., M, N)} array_like
+    A : {(M,), (..., M, N)} array_like
```
Comment (charris) [inline numpy/linalg/linalg.py:1817]: Can't say I'm happy about accepting 1-D vectors . . .

**Comment by charris** [comment]
Changing parameter names is risky, but I think we can get away with it here. I'll wait a bit in case someone else wants to comment.

**Comment by asmeurer** [comment]
Oh yeah I probably wasn't using main. Shouldn't I have gotten an error when trying to pull master?

**Comment by charris** [comment]
> Shouldn't I have gotten an error when trying to pull master?

`master` is probably still present in your remotes, exactly what is does I'm not sure.  Github has instructions for [updating local clones] (https://docs.github.com/en/github/administering-a-repository/managing-branches-in-your-repository/renaming-a-branch). I don't know if you have changed things in your github fork.

**Review by BvB93 — COMMENTED** [review]
You'll have have to update the parameter name in the stub file as well.
https://github.com/numpy/numpy/blob/923996a75a75fc5789a7495fcb14e4970e675bc6/numpy/linalg/__init__.pyi#L20

**Comment by asmeurer** [comment]
I guess git did give me an error when I pulled master, but it was very subtle about it. I wish GitHub would just alias master to main. I never know what branch I'm supposed to pull from anymore.

**Comment by charris** [comment]
Thanks Aaron.

**Pull request merged**